packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -386,16 +383,22 @@ export class Dialog extends LitElement implements OpenCloseComponent {
     this.calciteDialogClose.emit();
   }
 
-  private toggleDialog(value: boolean): void {
-    if (this.ignoreOpenChange) {
-      return;
+  private async toggleOpen(value: boolean): Promise<void> {
```

Comment:
This seems to be more about setting than toggling. Would something like `setOpenState`, `internalOpenSet` or something along those lines be a better fit? Applies to the `toggleClosed`.